### ISSUE TITLE
[BUG] spark.sql.legacy.parquet.datetimeRebaseModeInRead=CORRECTED or LEGACY still fails to read LEGACY date from parquet

### ISSUE THREAD

**Issue by viadea** [NVIDIA/spark-rapids#2277]
**Describe the bug**

No matter what is the value of `spark.sql.legacy.parquet.datetimeRebaseModeInRead` is, 
GPU Mode fails to read LEGACY date from parquet.

Another thing is regarding doc:
As per our Doc: https://nvidia.github.io/spark-rapids/docs/compatibility.html
It mentions:
```
For reads when spark.sql.legacy.parquet.datetimeRebaseModeInWrite is set to CORRECTED timestamps before the transition between the Julian and Gregorian calendars are wrong, but dates are fine. When spark.sql.legacy.parquet.datetimeRebaseModeInWrite is set to LEGACY, however both dates and timestamps are read incorrectly before the Gregorian calendar transition as described here.
```
Should we change `spark.sql.legacy.parquet.datetimeRebaseModeInWrite` to `spark.sql.legacy.parquet.datetimeRebaseModeInRead` here?

**Steps/Code to reproduce bug**
Please provide a list of steps or a code sample to reproduce the issue.
Avoid posting private or sensitive data.

```
import java.sql.Date
spark.conf.set("spark.rapids.sql.enabled", true)
spark.conf.set("spark.sql.legacy.parquet.datetimeRebaseModeInWrite", "LEGACY")
Seq(Date.valueOf("1500-12-25")).toDF("dt").write.format("parquet").mode("overwrite").save("/tmp/testparquet_legacy")

spark.conf.set("spark.sql.legacy.parquet.datetimeRebaseModeInRead", "LEGACY")
// or spark.conf.set("spark.sql.legacy.parquet.datetimeRebaseModeInRead", "CORRECTED")
spark.read.parquet("/tmp/testparquet_legacy").createOrReplaceTempView("date_legacy")
spark.sql("SELECT * FROM date_legacy").explain
spark.sql("SELECT * FROM date_legacy").show
```
Above fails with:
```
21/04/28 01:29:28 WARN TaskSetManager: Lost task 0.0 in stage 13.0 (TID 19) (111.111.111.111 executor 0): 
org.apache.spark.SparkUpgradeException: You may get a different result due to the upgrading of Spark 3.0: 
reading dates before 1582-10-15 or timestamps before 1900-01-01T00:00:00Z from Parquet files can be ambiguous, 
as the files may be written by Spark 2.x or legacy versions of Hive, which uses a legacy hybrid calendar that is different from Spark 3.0+'s Proleptic Gregorian calendar. 
See more details in SPARK-31404. The RAPIDS Accelerator does not support reading these 'LEGACY' files. 
To do so you should disable Parquet support in the RAPIDS Accelerator or set spark.sql.legacy.parquet.datetimeRebaseModeInRead to 'CORRECTED' to read the datetime values as it is.
```

**Expected behavior**

When spark.sql.legacy.parquet.datetimeRebaseModeInRead=CORRECTED or LEGACY, above read should work just like CPU Mode.

**Environment details (please complete the following information)**
 - Environment location: [Standalone, YARN, Kubernetes, Cloud(specify cloud provider)]
 - Spark configuration settings related to the issue

Rapids accelerator 0.5 snapshot
cuDF 0.19 snapshot
Spark 3.1.1 standalone

**Additional context**

**Comment by viadea** [comment]
I also found there is an issue open : https://github.com/NVIDIA/spark-rapids/issues/436 . But if this is the case, we should modify our Doc: https://nvidia.github.io/spark-rapids/docs/compatibility.html and also modify above error message.

**Comment by revans2** [comment]
This works as designed and we have tests for it [here](https://github.com/NVIDIA/spark-rapids/blob/branch-0.6/integration_tests/src/main/python/parquet_test.py) and documented [here](https://github.com/NVIDIA/spark-rapids/blob/branch-0.6/docs/compatibility.md#parquet), but the documentation is out of date and needs to be updated.

CUDF does not support the rebase functionality for either reads or writes. For writes if you set it to LEGACY we fall back to the CPU. For reads there is a really good chance that the data we read will not be in the bad range, so we just gamble and try to read it, then throw an exception if we run into a situation where it would be a problem. We have not prioritized fixing this because the vast majority of users do not have old dates, and if they do we will throw an exception instead of corrupting data.

**Comment by viadea** [comment]
Thanks @revans2 . 
I have tested reading LEGACY dates behaviors between GPU and CPU.
In summary, 
1. If the dates are newer than 1582-10-15, both CPU and GPU works fine, no ambiguous dates.
2. If the dates are between 1582-10-14 and 1582-10-05, both CPU and GPU show the same ambiguous dates = "original dates" + 10 days. 
3. If the dates are <= 1582-10-04, CPU can work fine. However GPU will start to fail. 

I understand this is the current situation/behavior, so I raised this PR to modify our Doc:
https://github.com/NVIDIA/spark-rapids/pull/2283/files

Please feel free to modify my PR if needed.thx a lot

**Comment by sameerz** [comment]
Closed based on PR #2283
